### DISCUSSION TITLE
Connect bleak client to GATT Server running in MicroPython on ESP32

### DISCUSSION THREAD

**Discussion by jlandgrave62** [hbldh/bleak#1601] [Q&A]
In this article : https://randomnerdtutorials.com/micropython-esp32-bluetooth-low-energy-ble/ 

The authors demonstrate how to create a BLE Server that publishes Read/Write Characteristics.  I would like to know if it's possible to use bleak to build a client app that runs on a small micropython remote with Two Buttons that, when pressed , update a characteristic on the ESP32 BLE Server.  

I didn't see any option in the bleak repository for the backend to be a microcontroller.  As far as I can tell the supported back ends are microprocessors running fat operating systems.  I'm trying to build a BLE Remote Client and Server using micropython on both devices where pressing buttons on the BLE client can set a value on the BLE Server.  The BLE Server then displays information on an LED Matrix based on the client BLE button that was pressed.

If bleak cannot handle this, I'd appreciate any references you can share that might allow this.

**Comment by dlech** [comment]
Bleak doesn't run on MicroPython, only Windows/Mac/Linux/Android. 

I would suggest asking about MicroPython at https://github.com/orgs/micropython/discussions. It has built-in BLE APIs.
